### PR TITLE
fig: init at 1.0.56

### PR DESCRIPTION
**Pull request by opeik**
###### Description of changes

Added package `fig` for macOS.

###### Things done

- Added `pkgs/development/tools/fig/default.nix`
- Added `fig` to `pkgs/top-level/all-packages.nix`

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/fig/default.nix [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  binname = "fig-darwin-universal";
+  version = "1.0.56";
+  revision = "414";
+  src = fetchurl {
+    name = "fig-${version}.dmg";
+    url = "https://versions.withfig.com/fig%20${revision}.dmg";
+    sha256 = "04cfvmxk187y4b0275b9mdy2y2c1vmvx4r95g0w0gwywzvz0p0rf";
+  };
+
+  meta = with lib; {
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fig/default.nix:20]: Meta should always be last.

File: pkgs/development/tools/fig/default.nix [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+, fetchurl
+, lib
+, makeWrapper
+, undmg
+, unzip
+}:
+stdenv.mkDerivation rec {
+  pname = "fig";
+  appname = "Fig";
+  binname = "fig-darwin-universal";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fig/default.nix:11]:
```suggestion
```

File: pkgs/development/tools/fig/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, fetchurl
+, lib
+, makeWrapper
+, undmg
+, unzip
+}:
+stdenv.mkDerivation rec {
+  pname = "fig";
+  appname = "Fig";
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fig/default.nix:10]:
```suggestion
```

File: pkgs/development/tools/fig/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    platforms = platforms.darwin;
+  };
+
+  sourceRoot = "${appname}.app";
+  nativeBuildInputs = [ makeWrapper undmg unzip ];
+  installPhase = ''
+    runHook preInstall
+    mkdir -p $out/{Applications/${appname}.app,bin}
+    cp -R . $out/Applications/${appname}.app
+    makeWrapper $out/Applications/${appname}.app/Contents/MacOS/${binname} $out/bin/${pname}
```
Comment (SuperSandro2000) [inline pkgs/development/tools/fig/default.nix:34]:
```suggestion
  sourceRoot = "Fig.app";
  nativeBuildInputs = [ makeWrapper undmg unzip ];
  installPhase = ''
    runHook preInstall
    mkdir -p $out/Applications/Fig.app $outbin
    cp -R . $out/Applications/Fig.app
    makeWrapper $out/Applications/Fig.app/Contents/MacOS/fig-darwin-universal $out/bin/fig
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

**Comment by arjan-s** [comment]
Fix also supports Linux now (in beta, but still). Would it be doable to integrate that into this package?

**Review by drupol — CHANGES_REQUESTED** [review]
Dear @opeik 

Do you plan to update this PR ?

Feel free to close it if you don't.

Thanks

**Comment by opeik** [comment]
Apologies, no, I don't. I ended up using the nix-darwin homebrew integration.

**Comment by drupol** [comment]
Fair enough, thanks for closing the PR.

**Pull request closed without merging**